packages/application-shell/src/apollo-links/header-link.js

```diff
@@ -13,9 +13,18 @@ const headerLink = new ApolloLink((operation, forward) => {
       `GraphQL target "${target}" is missing or is not supported`
     );
 
-  const projectKey = selectProjectKeyFromUrl();
-  // NOTE: keep header names with capital letters to avoid possible conflicts
-  // or problems with nginx.
+  /**
+   * NOTE:
+   *   The project key is read from the url in a project related appliation context.
+   *   This holds for most applications like `application-categories`, `application-discounts` etc.
+   *   However, the `application-account` does not run with the project key being part of the url.
+   *   As a result we allow passing the project key as a variable on the operation allowing
+   *   it to be the fallback.
+   */
+  const projectKey =
+    selectProjectKeyFromUrl() || operation.variables.projectKey;
```

Comment:
Should we do it the other way around? Prefer the `variables` option over the URL if given? Not 100% sure about it though...

Reply:
I would have expected the one from `variables` to take precedence as well.

Reply:
I would go the other way around too

Reply:
I had it like that before too. Wasn't sure either. I'll change it.